crates/witness-generator/src/rpc_generator.rs

```diff
@@ -196,3 +214,52 @@ impl RPCBlocksAndWitnesses {
         Ok(blocks_and_witnesses)
     }
 }
+
+/// Represents HTTP headers in a flat string format for easier configuration.
+///
+/// Each header should be in the format "key:value" or "key: value".
+#[derive(Debug, Clone, Default)]
+pub struct RpcFlatHeaderKeyValues {
+    headers: Vec<String>,
+}
+
+impl RpcFlatHeaderKeyValues {
+    /// Creates a new `RpcFlatHeaderKeyValues` from a vector of header strings.
+    ///
+    /// # Arguments
+    /// * `headers` - Vector of header strings in "key:value" format
+    pub fn new(headers: Vec<String>) -> Self {
+        Self { headers }
+    }
+}
+
+impl TryFrom<RpcFlatHeaderKeyValues> for HeaderMap {
+    type Error = anyhow::Error;
+
+    fn try_from(flat_headers: RpcFlatHeaderKeyValues) -> Result<Self, Self::Error> {
+        let header_pairs = flat_headers
+            .headers
+            .into_iter()
+            .map(|header| {
+                let (key, value) = header.split_once(':').ok_or_else(|| {
+                    anyhow::anyhow!("Invalid header format: '{header}'. Expected 'key:value'")
+                })?;
+
+                let name = HeaderName::from_str(key.trim())
+                    .map_err(|e| anyhow::anyhow!("Invalid header name '{key}': {e}"))?;
+
+                let value = HeaderValue::from_str(value.trim())
+                    .map_err(|e| anyhow::anyhow!("Invalid header value '{value}': {e}"))?;
+
+                Ok((name, value))
+            })
+            .collect::<Result<Vec<_>, Self::Error>>()?;
+
+        let mut header_map = HeaderMap::with_capacity(header_pairs.len());
+        for (name, value) in header_pairs {
+            header_map.insert(name, value);
+        }
+
+        Ok(header_map)
+    }
+}
```

Comment:
This was to simplify the client with the `try_into()` here:
![image](https://github.com/user-attachments/assets/9b341cd6-0988-4878-a250-2be4271eb799)